components/script/dom/subtlecrypto.rs

```diff
@@ -0,0 +1,328 @@
+            return promise;
+        }
+
+        let (task_source, canceller) = self
+            .global()
+            .as_window()
+            .task_manager()
+            .dom_manipulation_task_source_with_canceller();
+        let this = Trusted::new(self);
+        let trusted_promise = TrustedPromise::new(promise.clone());
+        let alg = normalized_algorithm.clone();
+        let _ = task_source.queue_with_canceller(
+            task!(generate_key: move || {
+                let subtle = this.root();
+                let promise = trusted_promise.root();
+                let key = match alg {
+                    Ok(NormalizedAlgorithm::AesKeyGenParams(key_gen_params)) => {
+                        subtle.generate_key_aes_cbc(key_usages, key_gen_params, extractable)
+                    },
+                    _ => Err(Error::NotSupported),
+                };
+                match key {
+                    Ok(key) => promise.resolve_native(&key),
+                    Err(e) => promise.reject_error(e),
+                }
+            }),
+            &canceller,
+        );
+
+        promise
+    }
+
+    /// <https://w3c.github.io/webcrypto/#SubtleCrypto-method-exportKey>
+    #[allow(unsafe_code)]
+    fn ExportKey(&self, format: KeyFormat, key: &CryptoKey, comp: InRealm) -> Rc<Promise> {
+        let promise = Promise::new_in_current_realm(comp);
+
+        let (task_source, canceller) = self
+            .global()
+            .as_window()
+            .task_manager()
+            .dom_manipulation_task_source_with_canceller();
+        let this = Trusted::new(self);
+        let trusted_key = Trusted::new(key);
+        let trusted_promise = TrustedPromise::new(promise.clone());
+        let _ = task_source.queue_with_canceller(
+            task!(export_key: move || {
+                let subtle = this.root();
+                let promise = trusted_promise.root();
+                let key = trusted_key.root();
+                let alg_name = key.algorithm();
+                if matches!(
+                    alg_name.as_str(), ALG_SHA1 | ALG_SHA256 | ALG_SHA384 | ALG_SHA512 | ALG_HKDF | ALG_PBKDF2
+                ) {
+                    promise.reject_error(Error::NotSupported);
+                    return;
+                }
+                if !key.Extractable() {
+                    promise.reject_error(Error::InvalidAccess);
+                    return;
+                }
+                let exported_key = match alg_name.as_str() {
+                    ALG_AES_CBC => subtle.export_key_aes_cbc(format, &*key),
+                    _ => Err(Error::NotSupported),
+                };
+                match exported_key {
+                    Ok(k) => {
+                        let cx = GlobalScope::get_cx();
+                        rooted!(in(*cx) let mut array_buffer_ptr = ptr::null_mut::<JSObject>());
+                        let _ = unsafe {
+                            ArrayBuffer::create(
```

Comment:
Are you able to use the safe [create_buffer_source](https://doc.servo.org/script/dom/bindings/buffer_source/fn.create_buffer_source.html) instead?